### PR TITLE
Endpoint for extract data from raster file

### PR DESCRIPTION
**Pull request by ManuelStardust**
## 🛠️ Changes
- Use get_areas_by_polygon endpoint for extracts stats from a raster file.

## 📝 Associated issues
#12

### CODE REVIEW

**Comment by cazapatamar** [comment]
I suggest to set the data estructure according to the current LP endpoint. For now, we can use a hardcoded value to set the id, something like `"id":"2011-2015"`.

imagen

**Comment by erikasv** [comment]
> I suggest to set the data estructure according to the current LP endpoint. For now, we can use a hardcoded value to set the id, something like `"id":"2011-2015"`.

@cazapatamar we'll make that change when the endpoint is completely implemented (all periods) and we're not gonna calculate percentages.

**Comment by erikasv** [comment]
I asked Julian to check the implementations to try to understand the differences, and hopefully we get an explanation as to why is necessary to use `~mask` instead of `mask` ( Manuel doesn't fully understand it).

Meanwhile It's necessary to return area values (ha) not only value counts

**Comment by erikasv** [comment]
Rebase this branch to `logs` and change this PR to that same branch

File: app/services/utils/raster.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        colormap={
+            0: (255, 0, 0, 255),
+            1: (128, 204, 102, 255),
+            2: (232, 214, 107, 255),
+        },
+    )
+
+
+def get_raster_values(raster_path, polygon):
+    crs = "EPSG:9377"
```
Comment (erikasv) [inline app/services/utils/raster.py:24]: where does this value come from?

File: app/services/utils/raster.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    categories = {"Perdida": 0, "Persistencia": 1, "No bosque": 2}
+
+    with Reader(raster_path) as cog:
+        data = cog.feature(polygon, dst_crs=crs)
+        transform = data.transform
+        mask = data.mask != 0
```
Comment (erikasv) [inline app/services/utils/raster.py:30]: what's the result of this operation? I guess is not a boolean

**Review by erikasv — CHANGES_REQUESTED** [review]
You're missing a toto for the tests of the areas values later.

Answer the other comments please, but they are non bloking

**Comment by cazapatamar** [comment]
I just compared the results of this endpoint to Biotablero's, and they are "consistent" and "similar." It is important to mention that there are significant differences between the polygons used and the raster resolutions.

Risaralda state (Biotablero vs. new endpoint):

    Perdida: 3587 vs. 1552
    Persistencia: 163660 vs. 162876
    No bosque: 188155 vs. 193478

File: app/services/utils/raster.py (hunk 2 above)
Comment (ManuelStardust) [reply app/services/utils/raster.py:30]: This operation removes the zeros present in the mask, converting them to False to filter the valid areas of the raster. [Creating our boolean mask](https://medium.com/@twelsh37/python-masking-data-before-plotting-6fd8de196aca)

File: app/services/utils/raster.py (hunk 1 above)
Comment (ManuelStardust) [reply app/services/utils/raster.py:24]: It's the plane coordinate projection that Julián recommended for areas in Colombia. [EPSG:9377
](https://epsg.io/9377)

**Comment by ManuelStardust** [comment]
> I just compared the results of this endpoint to Biotablero's, and they are "consistent" and "similar." It is important to mention that there are significant differences between the polygons used and the raster resolutions.
> 
> Risaralda state (Biotablero vs. new endpoint):
> 
> ```
> Perdida: 3587 vs. 1552
> Persistencia: 163660 vs. 162876
> No bosque: 188155 vs. 193478
> ```



> I just compared the results of this endpoint to Biotablero's, and they are "consistent" and "similar." It is important to mention that there are significant differences between the polygons used and the raster resolutions.

Yes, at this time differences are normal for several reasons that we must discuss, I'm compiling some of those considerations in a document. Additionally, a "To do" is added to this implementation for subsequent testing.

**Review by erikasv — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**